xbmc/video/dialogs/GUIDialogVideoVersion.cpp

```diff
@@ -268,15 +270,27 @@ void CGUIDialogVideoVersion::RefreshVideoVersionList()
   // get extras version list
   m_database.GetVideoVersions(itemType, dbId, *m_extrasVideoVersionList,
                               VideoVersionItemType::EXTRAS);
-  m_extrasVideoVersionList->SetContent(CMediaTypes::ToPlural(mediaType));
+  m_extrasVideoVersionList->SetContent(CMediaTypes::ToPlural(MediaTypeVideoVersion));
 
   // refresh extras version list
   CGUIMessage msg2(GUI_MSG_LABEL_BIND, GetID(), CONTROL_LIST_EXTRAS_VERSION, 0, 0,
                    m_extrasVideoVersionList.get());
   OnMessage(msg2);
 
-  // update default video version
-  m_database.GetDefaultVideoVersion(itemType, dbId, *m_selectedVideoVersion.get());
+  // update our default video version
+  UpdateDefaultVideoVersion();
+
+  // update our selected video version
+  UpdateSelectedVideoVersion();
+
+  // load artwork
+  CVideoThumbLoader loader;
```

Comment:
`CVideoThumbLoader` is an implementation of `CBackgroundInfoLoader`, the `void Load(CFileItemList& items)` function will populate images in a background thread. Keep it as a member variable of this class and make sure the thread is stopped when the window closes or another list is loaded with the below snippet.
```cpp
if (m_thumbLoader.IsLoading())
  m_thumbLoader.StopThread();
```

See how [CGUIWindowVideoBase](https://github.com/xbmc/xbmc/blob/master/xbmc/video/windows/GUIWindowVideoBase.cpp) uses `m_thumbLoader`, you probably don't need the `SetObserver` part. I'm not sure how easy it is to combine both lists you have here, but that will help loading the whole list in `ConvertVideoVersion`.

Reply:
Thanks @rmrector for this information. The artwork issue here is for video version dialog, there is not perf issue here as the items count is very limited. It is the original design issue with a unique_ptr for video version dialog, which cause issue when change the item (as reported in the related issues).

For ConvertVideoVersion artwork issue, with the feedback from forum, even if we fixed the artwork issue, trying to find a movie in a list of at least several hundred (or several thousand as reported in forum) has a usability issue. I'll create a new PR to fix that after this PR merged by only providing the detected same movies in the select dialog, instead of list all movies. User can always add a new version via "Add version" button in video version dialog to find the video file directly.